### PR TITLE
rabin karp pattern matching algo in cpp

### LINKED ISSUE #96: Rabin Karp Algorithm
Rabin-Karp algorithm is an algorithm used for searching/matching patterns in the text using a hash function. Unlike Naive string matching algorithm, it does not travel through every character in the initial phase rather it filters the characters that do not match and then performs the comparison.

This algorithm is available in the following language:
- C
- C++
- Java
- Python

### PR DESCRIPTION
**Pull request by vijay-kumar-yadav**
## 🛠️ Issue (Number)

Closes #96 

## 👨‍💻 Changes proposed
    Rabin Karp Algorithm in c++
    Readme updated

## ✔️ Check List (Check all the applicable boxes) 

- [ ✔️ ] My code follows the code style of this project.
- [✔️  ] This PR does not contain plagiarized content.
- [✔️  ] The title of my pull request is a short description of the requested changes.

## 📄 Note to reviewers

Apology for previous conduct I didn't know I have kept code from gfg as I have learnt this algo from gfg dsa course also I had written the code earlier when I was learning that too 1yr ago.

## 📷 Screenshots
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,7 +55,7 @@
 | 18.  | Binary Search Tree | [C](https://github.com/Kumar-laxmi/Algorithms/tree/main/C/Trees) | Trees |
 | 19.  | Red-Black Tree | [C++](https://github.com/Kumar-laxmi/Algorithms/blob/main/C%2B%2B/Trees/red_black.cpp), [Python](https://github.com/Kumar-laxmi/Algorithms/blob/main/Python/Trees/red_black.py) | Trees |
 | 20.  | Decision Tree | [Python](https://github.com/Kumar-laxmi/Algorithms/blob/main/Python/Machine%20Learning/Decision%20Tree/decision_tree.py) | Machine Learning |
-
+| 21.  | Rabin Karp | [C++](https://github.com/Kumar-laxmi/Algorithms/blob/main/C%2B%2B/Pattern-Matching/rabin_karp.cpp) | Pattern Matching |
```
Comment (Kumar-laxmi) [inline README.md:58]: Put Rabin Karp between index 8 & 9

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (Kumar-laxmi) [inline .vscode/settings.json:1]: please remove folder '.vscode'

**Review by Kumar-laxmi — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 2 above)
Comment (vijay-kumar-yadav) [reply .vscode/settings.json:1]: ok

File: README.md (hunk 1 above)
Comment (vijay-kumar-yadav) [reply README.md:58]: ok

**Comment by Kumar-laxmi** [comment]
@vijay-kumar-yadav Is it ready to merge?

**Comment by vijay-kumar-yadav** [comment]
> @vijay-kumar-yadav Is it ready to merge?
yes it is

**Comment by vijay-kumar-yadav** [comment]
> @vijay-kumar-yadav Is it ready to merge?

yes

**Comment by Kumar-laxmi** [comment]
@vijay-kumar-yadav Please resolve the **MERGE CONFLICT**

**Comment by vijay-kumar-yadav** [comment]
> @vijay-kumar-yadav Please resolve the **MERGE CONFLICT**

I have to open new pr then it will be fixed

**Pull request closed without merging**